kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizer.java

```diff
@@ -0,0 +1,31 @@
+package com.wepay.kafka.connect.bigquery.utils;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FieldNameSanitizer {
+
+  private static String sanitizeName(String name) {
+    name = name.replaceAll("^[^a-zA-Z_]+", "");
+    return name.replaceAll("[^a-zA-Z0-9_]", "_");
+  }
+
+
+  // Big Query specifies field name must begin with a letter or underscore and can only contain
+  // letters, numbers, and underscores
+  public static Map<String,Object> replaceInvalidKeys(Map<String, Object> map) {
```

Comment:
can we clarify how we would handle name collisions in this case? e.g. if someone has two keys, "01Mississippi" "02Mississippi" we'd now end up with the same key. I don't think we need to resolve collisions or do any fancy mangling, but we should at least clarify the behavior.

Reply:
we probably want a test for this as well, in case the code below throws an exception due to duplicate keys

Reply:
Solved!